### PR TITLE
HTTP and WS RPC APIs to listen on the same socket

### LINKED ISSUE #132: HTTP and WS RPC APIs to listen on the same socket
This was introduced in 0.9.43 - https://github.com/paritytech/substrate/pull/13384
The HTTP and WS APIs need to both listen on :9944. It was introduced into substrate with this PR: https://github.com/paritytech/substrate/pull/12663

### PR DESCRIPTION
**Pull request by ablax**
* Websocket configuration to differentiate http and ws request
 * Remove HttpRpc

Closes #132

### CODE REVIEW

**Review by vikinatora — CHANGES_REQUESTED** [review]
Please rename everything that is "ws" to "rpc". For example `WebSocketConfig` should be `RpcConfig`,

File: src/main/java/com/limechain/Main.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 import com.limechain.lightclient.LightClient;
-import com.limechain.rpc.ws.server.WebSocketRPC;
+import com.limechain.rpc.server.RpcApp;
 import sun.misc.Signal;
 
 public class Main {
     public static void main(String[] args) {
-        WebSocketRPC wsRpc = new WebSocketRPC();
+        RpcApp wsRpc = new RpcApp();
```
Comment (vikinatora) [inline src/main/java/com/limechain/Main.java]: `wsRpc` can be renamed to `rpcApp`

File: src/main/java/com/limechain/lightclient/LightClient.java [hunk 2]
```diff
@@ -18,12 +18,12 @@
 public class LightClient {
     // TODO: Add service dependencies i.e rpc, sync, network, etc.
     private final String[] cliArgs;
-    private final WebSocketRPC wsRpc;
+    private final RpcApp wsRpc;
```
Comment (vikinatora) [inline src/main/java/com/limechain/lightclient/LightClient.java]: `wsRpc` can be renamed to `rpcApp`

File: src/main/java/com/limechain/lightclient/LightClient.java [hunk 3]
```diff
@@ -18,12 +18,12 @@
...
     private final ConnectionManager connectionManager = ConnectionManager.getInstance();
     private Network network;
     private WarpSyncMachine warpSyncMachine;
 
-    public LightClient(String[] cliArgs, WebSocketRPC wsRpc) {
+    public LightClient(String[] cliArgs, RpcApp wsRpc) {
```
Comment (vikinatora) [inline src/main/java/com/limechain/lightclient/LightClient.java]: `wsRpc` can be renamed to `rpcApp`

File: src/test/java/com/limechain/lightclient/LightClientTest.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
 class LightClientTest {
     private LightClient lightClient;
 
-    private WebSocketRPC wsRpc;
+    private RpcApp wsRpc;
```
Comment (vikinatora) [inline src/test/java/com/limechain/lightclient/LightClientTest.java]: `wsRpc` can be renamed to `rpcApp`

Comment (vikinatora) [inline src/test/java/com/limechain/lightclient/LightClientTest.java]: `wsRpc` can be renamed to `rpcApp`

File: src/test/java/com/limechain/lightclient/LightClientTest.java [hunk 5]
```diff
@@ -29,7 +29,7 @@ private void setPrivateField(String fieldName, Object value)
 
     @BeforeEach
     public void setup() {
-        wsRpc = mock(WebSocketRPC.class);
+        wsRpc = mock(RpcApp.class);
```
Comment (vikinatora) [inline src/test/java/com/limechain/lightclient/LightClientTest.java]: `wsRpc` can be renamed to `rpcApp`

**Review by vikinatora — CHANGES_REQUESTED** [review]

**Review by vikinatora — APPROVED** [review]

**Review by Jess25051980 — APPROVED** [review]

**Review by Jess25051980 — APPROVED** [review]

**Pull request merged**